webroot/src/router/index.ts

```diff
@@ -46,8 +49,12 @@ router.beforeEach(async (to, from, next) => {
 
         if (!isLoggedIn) {
             next({ name: 'Logout' });
-        } else {
+        } else if ((isLicenseeRoute && isStaffRoute)
+        || (isLicenseeRoute && authStorage.getItem(tokens.licensee.AUTH_TYPE) === AuthTypes.LICENSEE)
+        || (isStaffRoute && authStorage.getItem(tokens.staff.AUTH_TYPE) === AuthTypes.STAFF)) {
```

Comment:
Similar question here. This is slightly confusing to me, because the all-caps `AUTH_TYPE` implies a constant to me.

Reply:
`tokens.staff.AUTH_TYPE` and `tokens.licensee.AUTH_TYPE` both evaluate to the constant `auth_type`.

This is a bit obfuscated with the current set up. Would the fix mentioned above "It might also make more sense to move the auth_type value up out of the staff / licensee objects, so just authStorage.getItem('auth_type') as this is more clear that this value is global?" cancel out your concern here?

Reply:
Yeah I think so - the more we can abstract the complexity here away into a single place, the better. And then, if a comment is still needed, we can add one in one place.